kevm-pyk/src/tests/integration/test_conformance.py

```diff
@@ -57,66 +62,97 @@ def _assert_exit_code_zero(pattern: Pattern) -> None:
     return set(slow_tests + failing_tests)
 
 
+def read_csv_file(csv_file: Path) -> tuple[tuple[Path, str], ...]:
+    with csv_file.open(newline='') as file:
+        reader = csv.reader(file)
+        return tuple((Path(row[0]), row[1]) for row in reader)
+
+
+ALL_VM_TESTS: Final = read_csv_file(ALL_VM_TESTS_CSV)
+ALL_BCHAIN_TESTS: Final = read_csv_file(ALL_BCHAIN_TESTS_CSV)
 SKIPPED_TESTS: Final = _skipped_tests()
 
-VM_TEST_DIR: Final = TEST_DIR / 'BlockchainTests/GeneralStateTests/VMTests'
-ALL_VM_TESTS: Final = tuple(VM_TEST_DIR.glob('*/*.json'))
-VM_TESTS: Final = tuple(test_file for test_file in ALL_VM_TESTS if test_file not in SKIPPED_TESTS)
-REST_VM_TESTS: Final = tuple(test_file for test_file in ALL_VM_TESTS if test_file in SKIPPED_TESTS)
+VM_TESTS: Final = tuple(test for test in ALL_VM_TESTS if test not in SKIPPED_TESTS)
+REST_VM_TESTS: Final = tuple(test for test in ALL_VM_TESTS if test in SKIPPED_TESTS)
+assert len(ALL_VM_TESTS) == len(VM_TESTS) + len(REST_VM_TESTS)
 
 
 @pytest.mark.parametrize(
-    'test_file',
+    'test_file, test_id',
     VM_TESTS,
-    ids=[str(test_file.relative_to(VM_TEST_DIR)) for test_file in VM_TESTS],
+    ids=[f'{test_file}:{test_id}' for test_file, test_id in VM_TESTS],
 )
-def test_vm(test_file: Path) -> None:
-    _test(test_file, 'DEFAULT', 'VMTESTS', 1, True)
+def test_vm(gst_data: dict[Path, dict[str, Any]], test_file: Path, test_id: str) -> None:
+    _test(gst_data, TEST_DIR / test_file, test_id, 'DEFAULT', 'VMTESTS', 1, True)
 
 
 @pytest.mark.skip(reason='failing / slow VM tests')
 @pytest.mark.parametrize(
-    'test_file',
+    'test_file, test_id',
     REST_VM_TESTS,
-    ids=[str(test_file.relative_to(VM_TEST_DIR)) for test_file in REST_VM_TESTS],
+    ids=[f'{test_file}:{test_id}' for test_file, test_id in REST_VM_TESTS],
 )
-def test_rest_vm(test_file: Path) -> None:
-    _test(test_file, 'DEFAULT', 'VMTESTS', 1, True)
+def test_rest_vm(gst_data: dict[Path, dict[str, Any]], test_file: Path, test_id: str) -> None:
+    _test(gst_data, TEST_DIR / test_file, test_id, 'DEFAULT', 'VMTESTS', 1, True)
 
 
-BCHAIN_TEST_DIR: Final = TEST_DIR / 'BlockchainTests/GeneralStateTests'
-ALL_BCHAIN_TESTS: Final = tuple(BCHAIN_TEST_DIR.glob('**/*.json'))
-BCHAIN_TESTS: Final = tuple(
-    test_file for test_file in ALL_BCHAIN_TESTS if test_file not in (list(SKIPPED_TESTS) + list(ALL_VM_TESTS))
-)
-REST_BCHAIN_TESTS: Final = tuple(
-    test_file for test_file in ALL_BCHAIN_TESTS if test_file in SKIPPED_TESTS and test_file not in ALL_VM_TESTS
-)
+EXCLUDED_TESTS = SKIPPED_TESTS.union(ALL_VM_TESTS)
+BCHAIN_TESTS: Final = tuple(test for test in ALL_BCHAIN_TESTS if test not in EXCLUDED_TESTS)
+REST_BCHAIN_TESTS: Final = tuple(test for test in ALL_BCHAIN_TESTS if test in EXCLUDED_TESTS)
+
+
+@pytest.fixture(scope='session')
+def gst_data() -> dict[Path, dict[str, Any]]:
+    all_gst_files: Final = {TEST_DIR / file for file, _ in ALL_BCHAIN_TESTS + ALL_VM_TESTS}
```

Comment:
```suggestion
    all_gst_files = [TEST_DIR / file for file, _ in ALL_BCHAIN_TESTS + ALL_VM_TESTS]
```

I assume each item should still be unique.